src/base/ceramicSafeWallet.ts

```diff
@@ -0,0 +1,102 @@
+import { MetaTransactionData } from "@safe-global/safe-core-sdk-types";
+import Safe, { EthersAdapter } from "@safe-global/protocol-kit";
+import { ethers, Wallet, HDNodeWallet } from "ethers";
+import { ceramicApiKit } from "./ceramicApiKit";
+import { concatSignatures, isTransactionSignedByAddress } from "./wallet";
+
+export interface TransactionPropose {
+  to: string;
+  value: bigint;
+  readyExecute: boolean;
+  safeTxHash: string;
+  txData: string;
+  operation: number;
+  signatures: string | null;
+}
+
+export class CeramicSafeWallet {
+  public address: string;
+  public signer: Wallet | HDNodeWallet;
+  private safeSdk: Safe;
+  private ceramicService: ceramicApiKit;
+
+  constructor(
+    address: string,
+    signer: Wallet | HDNodeWallet,
+    ceramicService: ceramicApiKit
+  ) {
+    this.address = address;
+    this.signer = signer;
+    this.ceramicService = ceramicService;
+  }
+
+  async connect(chainId: bigint) {
+    const ethAdapter = new EthersAdapter({
+      ethers,
+      signerOrProvider: this.signer,
+    });
+
+    this.safeSdk = await Safe.create({
+      ethAdapter: ethAdapter,
+      safeAddress: this.address,
+    });
+  }
+
+  async proposeTransaction(
+    transactions: MetaTransactionData[],
+    isProposor: boolean,
+    chainId: bigint
+  ): Promise<TransactionPropose | null> {
+    this.safeSdk ?? (await this.connect(chainId));
+    const nonce = await this.safeSdk.getNonce();
+    const tx = await this.safeSdk.createTransaction({ transactions, options: { nonce } });
```

Comment:
Writing a nonce at this step does not work for transaction generation.
see https://github.com/safe-global/safe-core-sdk/blob/main/packages/protocol-kit/src/utils/transactions/utils.ts#L45

Reply:
What I mean is that the following two implementations seem to work the same way.
1. const nonce = await this.safeSdk.getNonce();
    const tx = await this.safeSdk.createTransaction({ transactions, options: { nonce } });
2. const tx = await this.safeSdk.createTransaction({ transactions });

Reply:
correct，safeSDK already manages nonce for us
we can just read `nonce` from 
```
const tx = await this.safeSdk.createTransaction({ transactions, options: { nonce } });
tx.data.nonce
```

to me, I feel we should just use safeSdk internal nonce logic